_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -838,6 +835,27 @@ public function can_request( $request ) {
 			return current_user_can( 'jetpack_configure_modules' );
 		}
 	}
+
+	/**
+	 * Allows the `settings` and `module/<slug>` EDITABLE endpoints to accept both JSON and multi-part POST bodies.
+	 *
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 *
+	 * @return array|bool
+	 */
+	public function parse_settings_request_body( $request ) {
```

Comment:
Just call [`$request->get_params()`](https://core.trac.wordpress.org/browser/tags/4.7.2/src/wp-includes/rest-api/class-wp-rest-request.php#L437) instead, this will handle all parameter sources.

Then, since this function is only called once, you can inline the rest of the logic into `update_data`.  Something like this:

```php
$params = $request->get_params();
$params = array_filter( $params, 'is_string' );
unset( $params['context'] );
unset( $params['slug'] );
```